src/cli.rs

```diff
@@ -26,6 +26,25 @@ pub enum ReconstructSource {
 
 #[derive(Subcommand)]
 pub enum Command {
+    /// Download L2 state from L1 to JSON file.
+    #[command(hide = true)]
+    Download {
+        /// The Ethereum JSON-RPC HTTP URL to use.
+        #[arg(long)]
+        http_url: String,
+        /// Ethereum block number to start state import from.
+        #[arg(long, default_value_t=ethereum::GENESIS_BLOCK)]
+        start_block: u64,
+        /// The number of blocks to filter & process in one step over.
+        #[arg(long, default_value_t=ethereum::BLOCK_STEP)]
+        block_step: u64,
+        /// The number of blocks to process from Ethereum.
+        #[arg(long)]
+        block_count: Option<u64>,
```

Comment:
Should we maybe also have an arg for an explicit end block instead of going by count?

Reply:
Let's add it separately. I'll also separately improve the semantics of `block_count`.  Currently the `end_block` is a bit vague because of the filter batch processing (i.e. we often actually process over the computed `end_block` due to alignment of batch size). I just added this to have ability to perform variable length runs that also complete :slightly_smiling_face: